### DISCUSSION TITLE
How do i integrate the void tools everything into this?

### DISCUSSION THREAD

**Discussion by shadowdevnotreal** [dnGrep/dnGrep#676] [Q&A] [answered]
I am completely lost on how to only use one application versus having to run a search in void tools everything and dngrep.
Please tell me how to integrate the search feature into one application.

**Comment by doug24** [comment, marked as answer]
With both Everything and dnGrep installed, there should be a radio button on the right side of the file filters panel to use Everything:

<img src="https://user-images.githubusercontent.com/17227248/171746198-84add67f-c979-485e-ab34-cf85d327c95f.png" width="560"/>

In the box labeled "Everything search:" type any [filter you would use in Everything](https://www.voidtools.com/support/everything/searching/).  You can also use the other dnGrep file filters in addition to Everything filters - some like the date ranges are converted to Everything filters, and some are applied to the list of files returned by Everything.

 This generates a list of files to search, and dnGrep then searches the files using the search type and search pattern.

**Reply by shadowdevnotreal** [reply]
Much appreciated....it wasnt there before. Not sure why. I am going to need to play but this is amazing having both of these in one tool.
Think I may need dumpster diver for this search I am doing.......

**Reply by shadowdevnotreal** [reply]
Doug, quick update for you.
Thank you for that assist btw.
The main issue was I did not have everything actually running.
I have the service running at all times but the application itself needed to be running to be used in conjunction with dnGrep.
This is honestly an amazing tool (both).
Now I need to learn how to use the (filters) variables and I will be much better off.

**Reply by JVimes** [reply]
Doug, does it check for the Everything service, or the app? I assume the service is the real dependency, but I don't know.

**Reply by doug24** [reply]
Not sure if I have a clear answer.  As I understand it, the Everything service is responsible for keeping the file indexes up to date.  This service does not need to be running.

The Everything app does need to be running, but you don't need to have an Everything Search Window open.  You will see the Everything app in the collection of hidden app icons in the Task bar next to the clock.

dnGrep loads a native library Everything64.dll or Everything32.dll and pinvokes methods on this library.  The library makes the calls to the Everything service.   dnGrep will show Everything as an option if it can query this dll and get an Everything version number of 1.4.1 or higher.

**Reply by JVimes** [reply]
I see, thanks. The service is different than I assumed.

**Reply by shadowdevnotreal** [reply]
Yeah, I wasn't sure what the actual service did.
I had to make Everything launch at startup but did figure out the actual search window did NOT need to be open. The application itself does need to be open in some fashion, even if minimized.
I got it working and it is amazing.
Not sure how the whole using everything versus the other options, but for my needs it works great.
I had hundreds of PDF files and just right click on the folder and that opens dngrep right into the path I need, and I add my key search words below.
Now I see why linux users love this app so much.
